src/app/qgsoptions.cpp

```diff
@@ -2300,3 +2300,27 @@ void QgsOptions::scaleItemChanged( QListWidgetItem *changedScaleItem )
   addScaleToScaleList( changedScaleItem );
   mListGlobalScales->setCurrentItem( changedScaleItem );
 }
+
+double QgsOptions::zoomFactorValue()
+{
+  // Get the decimal value for zoom factor. This function is needed because the zoom factor spin box is shown as a percent value.
+  // The minimum zoom factor value is 1.01
+  if ( spinZoomFactor->value() == spinZoomFactor->minimum() )
+    return 1.01;
+  else
+    return spinZoomFactor->value() / 100.0;
+}
+
+void QgsOptions::setZoomFactorValue()
+{
+  // Set the percent value for zoom factor spin box. This function is for converting the decimal zoom factor value in the qgis setting to the percent zoom factor value.
+  if ( mSettings->value( QStringLiteral( "/qgis/zoom_factor" ), 2 ) <= 1.01 )
+  {
+    spinZoomFactor->setValue( spinZoomFactor->minimum() );
+  }
+  else
+  {
+    int percentValue = mSettings->value( QStringLiteral( "/qgis/zoom_factor" ), 2 ).toInt() * 100;
```

Comment:
ups I messed up here. it supposed to be `toDouble()`.
gonna create a patch for this.